borg/testsuite/archiver.py

```diff
@@ -276,8 +276,12 @@ def create_test_files(self):
                 os.path.join(self.input_path, 'hardlink'))
         # Symlink
         os.symlink('somewhere', os.path.join(self.input_path, 'link1'))
-        if xattr.is_enabled(self.input_path):
-            xattr.setxattr(os.path.join(self.input_path, 'file1'), 'user.foo', b'bar')
+        self.create_regular_file('fusexattr', size=1)
+        if not xattr.XATTR_FAKEROOT and xattr.is_enabled(self.input_path):
+            # ironically, due to the way how fakeroot works, comparing fuse
+            # file xattrs to orig file xattrs will FAIL if fakeroot supports
+            # xattrs, thus we only set the xattr if XATTR_FAKEROOT is False.
+            xattr.setxattr(os.path.join(self.input_path, 'fusexattr'), 'user.foo', b'bar')
```

Comment:
Because fakeroot with xattr-support does not propagate xattrs of the underlying file into "fakeroot space". Because the xattrs exposed by borgfs *are* these of an underlying file (from fakeroots point of view) they are invisible to the test process inside the fakeroot.